plugins/modules/azure_rm_subnet.py

```diff
@@ -267,6 +316,11 @@ def __init__(self):
                 type='str',
                 default='Enabled',
                 choices=['Enabled', 'Disabled']
+            ),
+            delegations=dict(
+                type='list',
+                elements='dict',
```

Comment:
I referred this doc : https://docs.ansible.com/ansible/latest/dev_guide/developing_program_flow_modules.html#elements

if we remove this line, it will go till Azure, and gives error like this : 
```
"msg": "Error creating or updating subnet test - Azure Error: InvalidServiceNameOnDelegation\nMessage: The service name Microsoft.CainerInstance/containerGroups specified on delegation /subscriptions/64caccf3-b508-41e7-92ed-d7ed95b32621/resourceGroups/test/providers/Microsoft.Network/virtualNetworks/test/subnets/test/delegations/test is not valid. Valid value(s) are: Microsoft.Web/serverFarms,Microsoft.ContainerInstance/containerGroups,Microsoft.Netapp/volumes,Microsoft.HardwareSecurityModules/dedicatedHSMs,Microsoft.ServiceFabricMesh/networks,Microsoft.Logic/integrationServiceEnvironments,Microsoft.Batch/batchAccounts,Microsoft.Sql/managedInstances,Microsoft.Web/hostingEnvironments,Microsoft.BareMetal/CrayServers,Microsoft.BareMetal/MonitoringServers,Microsoft.Databricks/workspaces,Microsoft.BareMetal/AzureHostedService,Microsoft.BareMetal/AzureVMware,Microsoft.BareMetal/AzureHPC,Microsoft.BareMetal/AzurePaymentHSM,Microsoft.StreamAnalytics/streamingJobs,Microsoft.DBforPostgreSQL/serversv2,Microsoft.AzureCosmosDB/clusters,Microsoft.MachineLearningServices/workspaces,Microsoft.DBforPostgreSQL/singleServers,Microsoft.DBforPostgreSQL/flexibleServers,Microsoft.DBforMySQL/serversv2,Microsoft.DBforMySQL/flexibleServers,Microsoft.ApiManagement/service,Microsoft.Synapse/workspaces,Microsoft.PowerPlatform/vnetaccesslinks,Microsoft.Network/managedResolvers,Microsoft.Kusto/clusters."
```

But if with this line, it will check possible values before hitting azure api , and error will be like this:
```
"msg": "value of serviceName must be one of: Microsoft.Web/serverFarms, Microsoft.ContainerInstance/containerGroups, Microsoft.Netapp/volumes, Microsoft.HardwareSecurityModules/dedicatedHSMs, Microsoft.ServiceFabricMesh/networks, Microsoft.Logic/integrationServiceEnvironments, Microsoft.Batch/batchAccounts, Microsoft.Sql/managedInstances, Microsoft.Web/hostingEnvironments, Microsoft.BareMetal/CrayServers, Microsoft.BareMetal/MonitoringServers, Microsoft.Databricks/workspaces, Microsoft.BareMetal/AzureHostedService, Microsoft.BareMetal/AzureVMware, Microsoft.BareMetal/AzureHPC, Microsoft.BareMetal/AzurePaymentHSM, Microsoft.StreamAnalytics/streamingJobs, Microsoft.DBforPostgreSQL/serversv2, Microsoft.AzureCosmosDB/clusters, Microsoft.MachineLearningServices/workspaces, Microsoft.DBforPostgreSQL/singleServers, Microsoft.DBforPostgreSQL/flexibleServers, Microsoft.DBforMySQL/serversv2, Microsoft.DBforMySQL/flexibleServers, Microsoft.ApiManagement/service, Microsoft.Synapse/workspaces, Microsoft.PowerPlatform/vnetaccesslinks, Microsoft.Network/managedResolvers, Microsoft.Kusto/clusters, got: Microsoft.CainerInstance/containerGroups found in delegations" 
```

Reply:
Yes, I see! Thanks!